js/main.js

```diff
@@ -7,63 +7,96 @@ var $zipCodeInput = document.querySelector('.zip-code');
 var $zipCodeTitle = document.querySelector('.zip-code-title');
 var $loading = document.querySelector('.fa-moon');
 var $headerBookIcon = document.querySelector('#book-header');
+var $headerStarIcon = document.querySelector('#star-header');
+var $headerMoonIcon = document.querySelector('#moon-header');
+var $headerPlusIcon = document.querySelector('#plus-header');
 var $footerBookIcon = document.querySelector('#book-footer');
-var $headerMoonIcon = document.querySelector('#header-moon');
 var $footerMoonIcon = document.querySelector('#moon-footer');
 var $footerPlusIcon = document.querySelector('#plus-footer');
+var $footerStarIcon = document.querySelector('#star-footer');
 var $observationView = document.querySelector('.observation-container');
 var $weatherView = document.querySelector('.container');
+var $favoriteView = document.querySelector('.favorite-container');
 var recordObv = document.querySelector('.observation-form');
 var $observationRecordView = document.querySelector('#observation-list');
 var $imageUrl = document.querySelector('.image-url');
 var $newUlobservations = document.querySelector('.list-observations');
-var observationId = null;
 var $observationTitle = document.querySelector('.observation-title');
 var $deleteEntryBtn = document.querySelector('.delete-entry');
 var $viewModal = document.querySelector('.view-modal');
 var $cancelbutton = document.querySelector('.cancel-button');
 var $confirmbutton = document.querySelector('.confirm-button');
 var $pastObvs = document.querySelector('.past-observations');
 var $searchInput = document.querySelector('#user-search');
+var $favoriteStarBtn = document.querySelector('#starBtn');
+var $emptystarBtn = document.querySelector('#emptystarBtn');
+var $favoriteImages = document.querySelector('#favorites');
+var $leftArrow = document.querySelector('.fa-chevron-left');
+var $rightArrow = document.querySelector('.fa-chevron-right');
```

Comment:
Are there any of these that are used exclusively in one function definition? You should only really make your DOM element variables global if you are going to use them more than once in more than one place. If any of these are used exclusively in one function, they would be better done as the first thing in whichever function definition they are used in.